mirrord/cli/src/config.rs

```diff
@@ -303,6 +316,175 @@ pub(super) struct ExecArgs {
     pub(super) binary_args: Vec<String>,
 }
 
+#[derive(Args, Debug)]
+#[command(group(ArgGroup::new("port-forward")))]
+pub(super) struct PortForwardArgs {
+    /// Target name to mirror.    
+    /// Target can either be a deployment or a pod.
+    /// Valid formats: deployment/name, pod/name, pod/name/container/name
+    #[arg(short = 't', long)]
+    pub target: Option<String>,
```

Comment:
I see that there a quite a few duplicate keys with `ExecConfig` maybe there is a case for doing 
```rust
#[derive(Args, Debug)]
pub(super) struct TargetParams {
    /// Target name to mirror.    
    /// Target can either be a deployment or a pod.
    /// Valid formats: deployment/name, pod/name, pod/name/container/name
    #[arg(short = 't', long)]
    pub target: Option<String>,

    /// Namespace of the pod to mirror. Defaults to "default".
    #[arg(short = 'n', long)]
    pub target_namespace: Option<String>,
}

impl TargetParams {
    pub fn as_env_vars(&self) -> Result<HashMap<String, OsString>, CliError> {
        let mut envs: HashMap<String, OsString> = HashMap::new();

        if let Some(target) = &self.target {
            envs.insert("MIRRORD_IMPERSONATED_TARGET".into(), target.into());
        }
        if let Some(namespace) = &self.target_namespace {
            envs.insert("MIRRORD_TARGET_NAMESPACE".into(), namespace.into());
        }
        
        envs
    }
}
```
And use it as 
```rust
#[clap(flatten)]
pub target: TargetParams,
```
Similar to how `ExecParams` is reused across `ExecArgs` and `ContainerArgs`

Reply:
ooh looks nice, will do 🫡